### ISSUE TITLE
Lexus gs450h canbus received but cant send 

### ISSUE THREAD

**Issue by aljini** [adamtheone/canDrive#24]
Hi there i have used your code and the interface 
I can receive canbus but when i send some back nothing happens 
Close the interface i can see on pycharm teminal ( wrong can package tx started )
I dont know whats the issue here

**Comment by aljini** [comment]
![Uploading 16460049422366135559964578633203.jpg…]()

**Comment by aljini** [comment]
![16460049875668623075985759149584](https://user-images.githubusercontent.com/58607810/155904794-071c6671-b1e3-4717-8e9f-dee62d02faf7.jpg)

**Comment by aljini** [comment]
![1646005215468472967837503308132](https://user-images.githubusercontent.com/58607810/155904911-ae476d4c-b056-48ff-bba1-90271900f598.jpg)

**Comment by SubhranshuSharma** [comment]
![Screenshot_2022-03-02_22_08_29](https://user-images.githubusercontent.com/84875500/156350466-f5a776fc-403b-4682-af8a-6aebac795972.png)
same

**Comment by SubhranshuSharma** [comment]
![Uploading 16462196707583895986704377252094.jpg…]()
Same

**Comment by SubhranshuSharma** [comment]
![16462197524707691360950157312861](https://user-images.githubusercontent.com/84875500/156351542-7009449e-a7c5-4eb7-9a6d-9dd0b2f16fd2.jpg)
Same

**Comment by aljini** [comment]
So its not working with you either did you try to send some messages back ?

**Comment by SubhranshuSharma** [comment]
If you mean sending packets into can bus, then still not successful

**Comment by adamtheone** [comment]
Hello.
We'll need more information to debug this. The worked for several other people, so I suspect either a hardware problem or something else. First, I highly recommend to try out the CAN examples codes just with the arduino. With that, you can make sure that your connection are fine and bus is working properly. Then, take a look at the ardunio raw serial output and see if that looks correct or not. Only try to debug the GUI as the last step, as the issue is somewhere before that, probably.

**Comment by aljini** [comment]
Everything seems fine in examples can recovering and sending and also i get rpm readings from the car so  on the Arduino side everything is fine 
But i suspect that there is a problem can drivers 
Or in the GUi the problem is i dont know how to debug that 
I really need it to work because i am converting my car to electric and this is process is needed in the conversion
So please help me fix that 
I have attached some photos of what errors i am getting 

![IMG_20220307_231824](https://user-images.githubusercontent.com/58607810/157110139-dfc54958-e4d3-4b39-9efa-048b5ab43dc6.jpg)

![IMG_20220307_234137](https://user-images.githubusercontent.com/58607810/157110162-533f9b9b-0cfb-4dd5-bdbc-63744595696c.jpg)

![IMG_20220307_234140](https://user-images.githubusercontent.com/58607810/157110184-f7e22d67-f6c6-4381-899b-014338c84e0c.jpg)

![IMG_20220307_234154](https://user-images.githubusercontent.com/58607810/157110221-1296f4e0-526f-406b-b382-3a7bc7997d2d.jpg)

![IMG_20220307_234533](https://user-images.githubusercontent.com/58607810/157110250-ca417984-c458-4955-8dce-7eedbab24155.jpg)

**Comment by aljini** [comment]
Today i found out that i can send some codes and the car recives it but the terminal says wrong paket and the sniffing stops it gives the command and the car recives it and it opens the door lock after that the sniffing stops and i cant recive or send anything to the car unless i disconnect it and connect it again 

Somehow i only got the door lock runing but the gui stops working i dont know why

**Comment by adamtheone** [comment]
The arduino serial output proves that it is not a GUI issue, because the packets are already corrupted when they exit the arduino. Are you sure that you are using the correct CAN bus speed?

**Comment by SubhranshuSharma** [comment]
> The arduino serial output proves that it is not a GUI issue, because the packets are already corrupted when they exit the arduino. Are you sure that you are using the correct CAN bus speed?

yes the can speed is 500e3 i know that as the engine rpm example works fine for me, is it possible that the car(i10 2013 in my case) doesnt have the can bus responsible for core functionality in the obd2 plug, is it possible that the cars brakes/stearing cant be controlled by the can bus on obd2 plug, another possibility is that the 500000 baud rate is not enough for 500e3 can speed so during python and arduino communication  some packets are missing, hence wrong package error.

**Comment by adamtheone** [comment]
I have no issues with 500mbaud, so I don't think that it is a bug in my code. It's absolutely possible that you cannot access certain modules in your car, a lot of newer vehicles have "firewalls" on the can bus to protect those devices from attackers. I'm not sure if this is the case in your situation.
